config/routes.rb

```diff
@@ -2,6 +2,17 @@
   crud = %i(index show create update destroy)
 
   namespace :v1 do
-    resources :projects, only: crud
+    resources :projects, only: crud do
+      post :create_task
+
+      member do
+        get :list_tasks
+      end
+
+    resources :tasks, only: [:update] do
+      put :update_task_state
+    end
```

Comment:
I do wonder though if you could discuss why you chose the routing scheme that you did.
I seems the routes you have created are:
```
POST /v1/projects/:project_id/create_task
GET /v1/projects/:id/list_tasks
PATCH /v1/projects/:project_id/tasks/:id
PUT /v1/projects/:project_id/tasks/:task_id/update_task_state
```

I might expect to see something a little more standard like:
Using nested routes
```
POST /v1/projects/:project_id/tasks
GET /v1/projects/:project_id/tasks
PATCH /v1/projects/:project_id/tasks/:id/update_state
```
Or using non-nested routes (downside, you need to pass `project_id` as a query param)
```
POST /v1/tasks
GET /v1/tasks
PATCH /v1/tasks/:id/update_state
```

What you have totally works, but I'm just curious on the thoughts behind your routing scheme choices.